Test/Invoke-WhiskeyExec.Tests.ps1

```diff
@@ -97,6 +99,22 @@ function GivenSuccessExitCode
     $script:successExitCode = $SuccessExitCode
 }
 
+function GivenTaskDefaultProperty
+{
+    param(
+        $Property
+    )
+
+    $regExMatches = Select-String -InputObject $Property -Pattern '([^\s"'']+)|"([^"]*)"|''([^'']*)''' -AllMatches

```

Comment:
Don't do this parsing. If other assertions needs the path/argument values, then update those assertions to take the expected path/arguments as parameters and have your tests pass them. There could be a bug hidden here. Explicit tests make it easier to spot when expections/assertions/preconditions don't line up.

Reply:
Good call, I've updated it so it's actually testing something..